components/SearchBar/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, useEffect } from "react";
+import Fuse from "fuse.js";
+import styled from "styled-components";
+import Projects from "@/components/Projects"; // Import your Projects component
+
+const StyledSearchBar = styled.div`
+  padding-top: 20px;
+`;
+
+export default function SearchBar({ projects, favourites, onToggleFavourite }) {
+  const [searchPattern, setSearchPattern] = useState("");
+  const [filteredProjects, setFilteredProjects] = useState([]);
+
+  useEffect(() => {
+    const fuseOptions = {
+      keys: ["title"],
+    };
+    const fuse = new Fuse(projects, fuseOptions);
+    if (searchPattern.trim() === "") {
+      setFilteredProjects(projects);
+    } else {
+      const results = fuse.search(searchPattern);
+      const filtered = results.map((result) => result.item);
+      setFilteredProjects(filtered);
+    }
+  }, [searchPattern, projects]);
```

Comment:
your code for filtering is good in general, but you don't need a second state and a useEffect. this results in unecessary rerenders. Please use a "derived state" here: 

```
export default function SearchBar({ projects, favourites, onToggleFavourite }) {
  const [searchPattern, setSearchPattern] = useState("");
  
  const fuseOptions = {
      keys: ["title"],
   };
   const fuse = new Fuse(projects, fuseOptions);

   const filteredProjects = fuse.search(searchPattern).map((result) => result.item);
    ...
```